### PR TITLE
cdpt 761 ct migrate to circleci credentials

### PR DESCRIPTION
**Pull request by MeenaModhvadia**
## Description
Migrate to short-lived credentials for circleci

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets
(https://dsdmoj.atlassian.net/browse/CDPT-761?atlOrigin=eyJpIjoiNDdlOTBmMjA2Y2Q1NDUzOWJhMDY0MTU5M2UxYjljY2QiLCJwIjoiaiJ9)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -134,38 +135,50 @@ references:
...
 
-  deploy_to_staging: &deploy_to_staging
+  deploy_to_dev: &deploy_to_dev
     run:
-      name: Deploy to Staging environment
+      name: Deploy to Development environment
       command: |
         build_tag=`cat /tmp/workspace/build_tag`
-        ./deploy.sh $build_tag staging circleci
+        ./deploy.sh $build_tag qa circleci
```
Comment (vertism) [inline .circleci/config.yml:181]:
```suggestion
        ./deploy.sh $build_tag development circleci
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -262,7 +270,32 @@ jobs:
       - slack/notify:
           color: '#1d990c'
           message: '${CIRCLE_USERNAME} deployed *${BUILD_TAG}* to *Development*'
-          webhook: ${SLACK_WEBHOOK_CT}
+          webhook: ${SLACK_WEBHOOK}
+
+  deploy_branch_to_dev: *do_dev_deployment_tasks
+  deploy_main_to_dev: *do_dev_deployment_tasks
```
Comment (vertism) [inline .circleci/config.yml:276]:
```suggestion
```

Comment (MeenaModhvadia) [reply .circleci/config.yml]: Thanks - removed

File: .circleci/config.yml (hunk 1 above)
Comment (MeenaModhvadia) [reply .circleci/config.yml]: Thanks - corrected

File: .circleci/config.yml [hunk 3]
```diff
@@ -262,7 +270,29 @@ jobs:
...
           message: '${CIRCLE_USERNAME} deployed *${BUILD_TAG}* to *Development*'
-          webhook: ${SLACK_WEBHOOK_CT}
+          webhook: ${SLACK_WEBHOOK}
+
+jobs:
+
+  build_and_test:
+    <<: *test_container_config
+    <<: *defaults
+
```
Comment (vertism) [inline .circleci/config.yml:280]: Add this back in to speed up the testing section
```suggestion
    resource_class: xlarge
    
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -196,60 +218,48 @@ references:
         echo "export BUILD_TAG=${build_tag}" >> $BASH_ENV
         source $BASH_ENV
 
-jobs:
+  jobs:
```
Comment (vertism) [inline .circleci/config.yml:221]:
```suggestion
jobs:
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -196,60 +218,48 @@ references:
...
-      - *set_up_the_database
-      - *rubocop
-      - *update_chromedriver
-      - *setup_test_reporter
-      - *run_unit_and_feature_tests
-      - *upload_test_coverage
-
-  build_branch_and_push_to_ecr:
-    <<: *deploy_container_config
+    build_branch_and_push_to_ecr:
```
Comment (vertism) [inline .circleci/config.yml:223]:
```suggestion
  build_branch_and_push_to_ecr:
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -197,75 +218,84 @@ references:
...
+        - *install_psql
+        - *install_dependencies
+        - *save_cache
+        - *wait_for_db
+        - *set_up_the_database
+        - *rubocop
+        - *update_chromedriver
+        - *setup_test_reporter
+        - *run_unit_and_feature_tests
+        - *upload_test_coverage
```
Comment (vertism) [inline .circleci/config.yml:239]:
```suggestion
      - checkout
      - *restore_cache
      - *install_bundler
      - *check_bundler_version
      - *install_psql
      - *install_dependencies
      - *save_cache
      - *wait_for_db
      - *set_up_the_database
      - *rubocop
      - *update_chromedriver
      - *setup_test_reporter
      - *run_unit_and_feature_tests
      - *upload_test_coverage
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -366,27 +399,27 @@ workflows:
...
+      # - deploy_main_to_qa:
+      #     requires:
+      #       - deploy_main_to_qa_approval
+      # - deploy_main_to_demo_approval:
+      #     type: approval
+      #     requires:
+      #       - deploy_main_to_demo
+      # - deploy_main_to_demo:
+      #     requires:
+      #       - deploy_main_to_demo_approval
```
Comment (vertism) [inline .circleci/config.yml:415]: I don't think this needs to change, you can revert it back to the original code on the left.

File: .circleci/config.yml [hunk 8]
```diff
@@ -366,27 +399,27 @@ workflows:
...
 
   build_and_deploy_branch:
     jobs:
       - build_and_test:
-          filters:
-            branches:
-              ignore: main
+        filters:
+          branches:
+            ignore: main
```
Comment (vertism) [inline .circleci/config.yml:422]: This doesn't need to change, it's correctly indented on the left.

**Review by lucas-shaw — APPROVED** [review]
LGTM

**Review by vertism — CHANGES_REQUESTED** [review]
A couple of indent issues to deal with

File: .circleci/config.yml [hunk 9]
```diff
@@ -197,75 +218,84 @@ references:
...
+        - attach_workspace:
+            at: /tmp/workspace
+        - *aws_setup
+        - *ecr_login
+        - *configure_build_tag
+        - *build_and_push_image
+        - persist_to_workspace:
+            root: workspace
+            paths:
+              - build_tag
```
[the remark this answers is not in the document]
Comment (vertism) [reply .circleci/config.yml:271]: @MeenaModhvadia I think this is the last thing that needs updating :)

File: .circleci/config.yml [hunk 10]
```diff
@@ -219,53 +239,63 @@ jobs:
...
-      - *build_and_push_docker_image
+      - *aws_setup
+      - *ecr_login
+      - *configure_build_tag
+      - *build_and_push_image
       - persist_to_workspace:
           root: workspace
           paths:
-            - build_tag
+          - build_tag
```
Comment (vertism) [inline .circleci/config.yml:255]:
```suggestion
            - build_tag
```

**Review by vertism — APPROVED** [review]

**Pull request merged**